src/main/kotlin/dev/reimer/tex/gradle/plugin/task/TexCompile.kt

```diff
@@ -43,17 +45,17 @@ open class TexCompile : FilteredSourceTask(), TexScope {
      */
     @get:Internal
     val buildDir = project.directoryProperty {
-        project.buildDir.resolve(TexPlugin.EXTENSION_NAME).resolve(name)
+        project.layout.buildDirectory.get().asFile.resolve(TexPlugin.EXTENSION_NAME).resolve(name)
     }
 
     init {
         val generateResources = let { tex ->
             project.tasks.register<TexResources>(
-                "generateResourcesFor${name.capitalize()}"
+                "generateResourcesFor${name.replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString() }}"
```

Comment:
Why do we need this complicated renaming? Doesn't `capitalize` work just fine?

Reply:
Ah, thanks for the clarification :+1:  I didn't know Kotlin deprecated that function. I agree, that `.replaceFirstChar{it.replaceFirstChar(Char::uppercase)}` looks most elegant and should do the job while being readable.

Reply:
```suggestion
                "generateResourcesFor${name.replaceFirstChar(Char::titlecase)}"
```